### ISSUE TITLE
Angular Elements: Unexpected Behaviour of Web Component's styles ( + best practices of writing styles for web components)

### ISSUE THREAD

**Issue by dali-trabelsi** [angular/angular#48078]
### Which @angular/* package(s) are the source of the bug?

elements

### Is this a regression?

No

### Description

I am using Angular Elements to create and export multiple web components to be used on other websites.
My main issue is that the global styles in the parent web app seem to be overriding styles from my web components particularly for native elements like <input/> and <p/>. 
So I was wondering if this behavior is normal and in that case what can be done to encapsulate and make all my web component's styles take priority. 
Also when I name my classes something like "alert", it gets overridden by bootstrap's class names from the parent webapp. 
Are there any best practices when writing styles for web components which differs from the usual workflow when developing a web app?

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 14.1.3
Node: 16.18.0
Package Manager: npm 8.19.2 
OS: win32 x64

Angular: 14.1.2
... animations, cdk, common, compiler, compiler-cli, core
... elements, forms, platform-browser, platform-browser-dynamic
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1401.3
@angular-devkit/build-angular   14.1.3
@angular-devkit/core            14.1.3
@angular-devkit/schematics      14.1.3
@angular/cli                    14.1.3
@schematics/angular             14.1.3
rxjs                            7.5.6
typescript                      4.7.4
```

**Comment by JeanMeche** [comment]
Did you build your components using [ViewEncapsulation.ShadowDom](https://angular.io/api/core) ?

**Comment by dali-trabelsi** [comment]
> Did you build your components using [ViewEncapsulation.ShadowDom](https://angular.io/api/core) ? This options should normally be enough to make them independent from the parents !



> Did you build your components using [ViewEncapsulation.ShadowDom](https://angular.io/api/core) ? This options should normally be enough to make them independent from the parents !

Correct, I wasn't aware of that option and assumed it would be encapsulated automatically. Thanks @JeanMeche !
